### ISSUE TITLE
有关loss变为nan的情况，我看了之前的解答，但还是想问问

### ISSUE THREAD

**Issue by xiiiiiiii** [zhiqwang/image-captioning#18]
我要用mobilenetv2+ctc训练一批自己的数据，数据的size是32*258,数据集图片是这样的，
![image](https://user-images.githubusercontent.com/45999214/59012527-5544e480-886a-11e9-9235-60789fe9fbaf.png)
都是32*258,一共9k张训练，1k张验证，标签是这样的：
00000000.jpg 144 80 91 9 213 24 16 217 91 682 129 100 5 
00000001.jpg 140 481 9 102 2612 31 330 71 65 15 4 
00000002.jpg 1688 195 91 49 678 4 24 1166 2700 58 135 
每一张图片中的字都是不一定的，是10个左右，比如11,13这样

首先，我在代码中改了：
**parser.add_argument('--width', type=int, default=256,）**
然后运行的代码为：
 **python main.py --gpu-id 0 --not-pretrained --optimizer adam**
可是在运行了四个epoch后，出现了loss为nan的情况

ctc设置为：
**criterion = nn.CTCLoss(zero_infinity=True)**
图片的方差和标准差设置了成自己图片的方差和标准差，为：
**model_params['mean'] = (0.57680161,0.57680161,0.57680161)
model_params['std'] = (0.1311234,0.1311234,0.1311234)**
![image](https://user-images.githubusercontent.com/45999214/59012498-3fcfba80-886a-11e9-81de-6088e28ebf58.png)
求问，这是什么原因啊，我按您之前讲解的都设置好了，还是出现了nan的情况，因为我得训练mobilenetv2的网络因为课题需要，谢谢您

**Comment by zhiqwang** [comment]
如果只是测试，使用 `densenet_cifar` 这个目前得到的结果怎么样呢?

**Comment by xiiiiiiii** [comment]
我还是有这两个小问题，第一个就是我的数据集中每张图片的字数不是固定的，这种情况的话用不用对数据集做什么处理，比如要不要搞成每张图片的标签是一样的这种。
第二个问题就是我看了pytorch中的那个关于ctcloss解决方案的讨论，我就看到他说在maser里把代码更新了，但是没有看到说是在nightly版本里更新了，这个是从哪里看到的，怎么知道是nightly版本里对ctc更新了呀，我总觉得我下的版本不对，是在官网里下载nightly版本的pytorch吗？
我知道其实没有人有义务来回答我，我也问别人了，但是解决方法都不太对，所以可以拜托解答下这个问题吗？我保证以后的问题自己想，因为我今天看了好几遍有关ctc更新的内容，都没有发现他说他是在更新到哪个版本里了，让我很费解
 
---原始邮件---
发件人: "Zhiqiang Wang"<notifications@github.com>
发送时间: 2019年6月6日(星期四) 下午3:50
收件人: "zhiqwang/crnn.pytorch"<crnn.pytorch@noreply.github.com>;
抄送: "guopeijun"<494207346@qq.com>;"Author"<author@noreply.github.com>;
主题: Re: [zhiqwang/crnn.pytorch] 有关loss变为nan的情况，我看了之前的解答，但还是想问问 (#18)



如果只是测试，使用 densenet_cifar 这个目前得到的结果怎么样呢?

**Comment by zhiqwang** [comment]
1. Nope, datasets support this form
2. the nightly pytorch kernel version update everyday I guess, you can test it by reinstall `pytorch_nightly`

**Comment by zhiqwang** [comment]
BTW, Where you use the method of `clib` optim?

**Comment by xiiiiiiii** [comment]
> BTW, Where you use the method of `clip` optim?
in main.py  的 train函数里 
`loss = criterion(log_probs, targets, input_lengths, target_lengths)
 losses.update(loss.item())
 loss.backward()
 nn.utils.clip_grad_norm(parameters, max_norm, norm_type=2)
 optimizer.step()`

但是损失不下降，也不上升，一直在2左右，btw，i use mobilenetv2,
and today,i watch another way, put nan or inf gradient is zero ,today i try it,but it has not finished training yet,i don`t konw how effective it is

**Comment by xiiiiiiii** [comment]
> BTW, Where you use the method of `clip` optim?

在 main.py 的 train函数里
loss = criterion(log_probs, targets, input_lengths, target_lengths) 
losses.update(loss.item()) 
loss.backward() 
nn.utils.clip_grad_norm(parameters, max_norm, norm_type=2) 
optimizer.step()

但是损失不下降，也不上升，一直在2左右，btw，i use mobilenetv2,
and today,i watch another way, put nan or inf gradient is zero ,today i try it,but it has not finished training yet,i don`t konw how effective it is
